public/js/NeighbourPlot.js

```diff
@@ -0,0 +1,199 @@
+var d3 = require('d3');
+require('d3-tip')(d3);
+
+/**
+ * Draws PCA scatterplot of the data Clicking on scatterplot points will
```

Comment:
I've been reading up on tSNE. Looks great. I think a nice addition would be the option to give the user a few options about the dimension is reduced to draw the scatterplot. A dropdown box with PCA, MDS, tSNE or any others are are applied.

Regarding the docstrings, how strict we adhere to the JSDoc syntax all depends on whether or not we want to parse them. I don't think we will, so I'm happy to add a little more detail as you've described above.